### PR TITLE
Set up demo with Docker

### PR DESCRIPTION
**Pull request by apedroferreira**
Possible solution for https://github.com/mui/mui-toolpad/issues/1056

### CODE REVIEW

**Comment by apedroferreira** [comment]
@Janpot would this be a correct way to do things using Docker?
I tried to deploy https://dashboard.render.com/web/srv-ccu728t3t398codsgi90/events based on https://github.com/mui/mui-toolpad/pull/426 but it seems to run out of memory too.

**Comment by Janpot** [comment]
This wouldn't use out docker image. It's rebuilding it using our dockerfile. 🤔 Perhaps we need to create a dockerfile that has the desired image as base?

```docker
# Dockerfile.demo
FROM muicom/toolpad:123455
```

**Comment by apedroferreira** [comment]
Got it, I wasn't fully understanding last week that we wanted to use the remote image - that should be great, I'll try it out, probably on a separate repo.

**Comment by apedroferreira** [comment]
@Janpot I tried Render with the remote image and still got an out of memory error https://dashboard.render.com/web/srv-cd26q32en0hm1tsfqoh0
I wonder if I did something wrong or if our memory problems could be elsewhere?

**Comment by Janpot** [comment]
Ah right, we're hitting https://github.com/prisma/prisma/issues/11413 again

File: docker/images/toolpad/Dockerfile.demo [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM muicom/toolpad:latest
```
Comment (Janpot) [inline docker/images/toolpad/Dockerfile.demo:1]: Shall we pin the version

Comment (apedroferreira) [reply docker/images/toolpad/Dockerfile.demo]: I pinned it to v0.0.21, will just have to manually keep it updated.
Other than that does this flow with the `demo` branch look good to you if it works? Or would you still prefer we use a separate repo for the demo?

**Comment by oliviertassinari** [comment]
A bit of topic, I just saw: 

Screenshot 2022-10-20 at 15 13 11

https://app.forestadmin.com/Live%20Demo/Production/Operations/data/806052/index

It's could be interesting to have something like this (I haven't seen the demo app UX yet, and I changed the color contrast because OMG dark seriously!)

File: render.yaml [hunk 2]
```diff
@@ -27,11 +27,38 @@ services:
         - docs/**
         - '**/*.spec.ts'
 
+  - type: web
+    name: toolpad-demo
+    env: docker
+    branch: demo
+    dockerfilePath: ./docker/images/toolpad/Dockerfile.demo
+    dockerContext: ./
```
Comment (Janpot) [inline render.yaml:35]: Perhaps we can set the context to 
```yml
dockerContext: ./docker/images/toolpad/
```
? That should speed up the build as it doesn't have to upload the whole repository as a context. The docker file doesn't (and shouldn't) use anything from the repository

Comment (apedroferreira) [reply render.yaml]: I'll keep the path just in case?

File: render.yaml [hunk 3]
```diff
@@ -27,11 +27,38 @@ services:
...
+  - type: web
+    name: toolpad-demo
+    env: docker
+    branch: demo
+    dockerfilePath: ./docker/images/toolpad-demo/Dockerfile
+    dockerContext: ./docker/images/toolpad-demo/
+    healthCheckPath: /health-check
+    plan: starter plus
+    pullRequestPreviewsEnabled: false
+    envVars:
```
Comment (Janpot) [inline render.yaml:39]: Shouldn't we set the demo mode env var?

Comment (apedroferreira) [reply render.yaml:39]: Yeah good one, I'll set every env var except the secret ones

Comment (apedroferreira) [reply render.yaml:39]: I guess just `TOOLPAD_DEMO` should do

Comment (Janpot) [reply render.yaml:39]: Yep, as much infrastructure expressed in code as possible

**Review by Janpot — APPROVED** [review]

**Pull request merged**